src/Data/HashMap/Strict/InsOrd.hs

```diff
@@ -400,8 +412,65 @@ mapWithKey f (InsOrdHashMap i m) =
   where
     f' k (P j x) = P j (f k x)
 
+foldMapWithKey :: Monoid m => (k -> a -> m) -> InsOrdHashMap k a -> m
+foldMapWithKey f = foldMap (uncurry f) . toList
```

Comment:
Why not just use `ifoldMap` here? If this one is better, why not define `ifoldMap = foldMapWithKey`?